### DISCUSSION TITLE
Templated HTML attributes

### DISCUSSION THREAD

**Discussion by nexovec** [a-h/templ#955] [Ideas]
Would there be a way to make something like `<button class="circle" @inlineJSCallbackOnClick>A</button>` work?

Most straightforward thing I can think of is just enabling this for a fuction that returns a safe string, something like

```templ
func InlineOnclick(url string) templ.SafeString {
  return fmt.Sprintf("onclick=%s", url)
}

templ Button(){
  <button class="circle" @InlineOnClick("/logout")>A</button>
}
```

**Comment by joerdav** [comment]
Hey!

I think the best bet here would be to use the spread attribute feature. Your function can return the type `templ.Attributes` and can be called like:

```templ
<button class="circle" { InlineOnClick("/logout")... }>A</button>
```

https://templ.guide/syntax-and-usage/attributes#spread-attributes
